src/preset.ts

```diff
@@ -1483,79 +1469,34 @@ function getConfigurePresetForPresetHelper(folder: string, preset: BuildPreset |
-        return null;
-    }
-
-    // Expand strings under the context of current preset
-    const expandedPreset: BuildPreset = { name };
-    const expansionOpts: ExpansionOptions = await getExpansionOptions(workspaceFolder, sourceDir, preset);
-
-    // Expand environment vars first since other fields may refer to them
-    if (preset.environment) {
-        expandedPreset.environment = EnvironmentUtils.createPreserveNull();
-        for (const key in preset.environment) {
-            if (preset.environment[key]) {
-                expandedPreset.environment[key] = await expandString(preset.environment[key]!, expansionOpts, errorHandler);
-            }
-        }
-    }
-
-    expansionOpts.envOverride = expandedPreset.environment;
+    const preset = getBuildPresetInheritsImpl(folder, name, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, allowUserPreset, configurePreset, enableTryApplyDevEnv, errorHandler, inheritedByPreset);
+    errorHandlerHelper(name, errorHandler);
 
-    // Expand other fields
-    if (preset.targets) {
-        if (util.isString(preset.targets)) {
-            expandedPreset.targets = await expandString(preset.targets, expansionOpts, errorHandler);
-        } else {
-            expandedPreset.targets = [];
-            for (let index = 0; index < preset.targets.length; index++) {
-                expandedPreset.targets[index] = await expandString(preset.targets[index], expansionOpts, errorHandler);
-            }
-        }
-    }
-    if (preset.nativeToolOptions) {
-        expandedPreset.nativeToolOptions = [];
-        for (let index = 0; index < preset.nativeToolOptions.length; index++) {
-            expandedPreset.nativeToolOptions[index] = await expandString(preset.nativeToolOptions[index], expansionOpts, errorHandler);
-        }
-    }
-
-    if (preset.condition) {
-        expandedPreset.condition = await expandCondition(preset.condition, expansionOpts, errorHandler);
-    }
-
-    // Other fields can be copied by reference for simplicity
-    merge(expandedPreset, preset);
-
-    errorHandlerHelper(preset.name, errorHandler);
-
-    return expandedPreset;
+    return preset;
 }
 
-async function expandBuildPresetImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler, inheritedByPreset?: BuildPreset): Promise<BuildPreset | null> {
+async function getBuildPresetInheritsImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler, inheritedByPreset?: BuildPreset): Promise<BuildPreset | null> {
     let preset = getPresetByName(buildPresets(folder), name);
```

Comment:
I think we need to modify `buildPresets(folder)` to also account for `enableTryApplyDevEnv`, similar to how we do for configure presets. 

Right now, when a user has a build preset defined in the presets file, and then they go to select it, we get to this line and the `preset.__expanded` is `true`, and therefore we don't do any of the `getBuildPresetInheritsHelper` logic, because it exits early. This is likely true for all other types of presets as well

Reply:
I just tested with `testPresets` as well and confirmed my suspicions

Reply:
done